packages/graphql/src/classes/Neo4jGraphQL.ts

```diff
@@ -284,23 +285,23 @@ class Neo4jGraphQL {
             jwtPayloadFieldsMap: this.jwtFieldsMap,
         };
 
-        const resolversComposition = {
-            "Query.*": [wrapQueryAndMutation(wrapResolverArgs)],
-            "Mutation.*": [wrapQueryAndMutation(wrapResolverArgs)],
-        };
+        const queryResolvers = wrapQueryFields(this.schemaModel, [wrapQueryAndMutation(wrapResolverArgs)]);
+        const mutationResolvers = wrapMutationFields(this.schemaModel, [wrapQueryAndMutation(wrapResolverArgs)]);
+        const subscriptionResolvers = {};
 
         if (this.features.subscriptions) {
-            resolversComposition["Subscription.*"] = wrapSubscription({
+            const wrapSubscriptionResolverArgs = {
                 subscriptionsEngine: this.features.subscriptions,
                 schemaModel: this.schemaModel,
                 authorization: this.authorization,
                 jwtPayloadFieldsMap: this.jwtFieldsMap,
-            });
+            };
+            subscriptionResolvers["Subscription.*"] = [wrapSubscription(wrapSubscriptionResolverArgs)];
         }
 
         // Merge generated and custom resolvers
         const mergedResolvers = mergeResolvers([...asArray(resolvers), ...asArray(this.resolvers)]);
-        return composeResolvers(mergedResolvers, resolversComposition);
+        return composeResolvers(mergedResolvers, { ...queryResolvers, ...mutationResolvers, ...subscriptionResolvers });
```

Comment:
It feels like this implementation is more complex than it needs to be. It seems like the `wrapQueryAndMutation` function is now differentiating between our generated resolvers and custom resolvers, and you have had to add a lot of metadata to the schema model to support this. But we already have this knowledge in this function. So we're merging the resolvers together just to then split them apart again. Happy to discuss, let me know if you want to talk it through!